docs/CODING_STYLE.rst

```diff
@@ -62,37 +67,188 @@ Coding guide
    -  Declare variables where needed (not all at function start)
    -  Use the smallest required scope
    -  Variables in a file (outside functions) are always *static*
-   -  Do not use global variables (use functions to set/get static
-      variables)
+   -  Do not use global variables (use functions to set/get static variables)
 
 Comments
 --------
+Before every function prototype in ``.h`` files, include a Doxygen-formatted comment
+like this:
+
+.. code-block:: c
+
+    /**
+     * Brief description.  Add a blank line + additional paragraphs when more detail is needed.
+     * @param  parent     brief description of argument.  Additional detail can appear
+     *                    on subsequent lines.  List of accepted values:
+     *                    - value one
+     *                    - value two
+     *                    - value three
+     * @return   brief description of return value.
+     */
+    type_name_t * lv_function_name(lv_obj_t * parent);
+
+The normal comment prefix ``/**`` causes the comment to document the code member
+coming *after* the comment.  When documenting a code member that is *before* the
+comment, such as a struct member, use ``/**<`` like this:
+
+.. code-block:: c
 
-Before every function have in ``.h`` files a comment like this:
+    /**
+     * Brief description of struct
+     *
+     * When more detail is needed, add a blank line then the detail.
+     */
+    typedef struct _lv_example_type_t {
+        char      *text;    /**< Brief description of this member */
+        uint16_t   length;  /**< Brief description of this member */
+    } lv_example_type_t;
```

Comment:
> What do you think?

Consistency is always a good thing.  I overlooked it because I didn't realize that was a sort of agreement in play about how structs were declared, and I free-typed that and pulled what came out of my head.  :-)  Good catch, and subtle!